### PR TITLE
chore: upgrade rollup and plugins, change typescript compiler to `tsc` and add tree shaking

### LINKED ISSUE #861: Upgrade to build script for react package
Upgrade to Rollup 3 and clean up build configuration for our react package.

https://rollupjs.org/migration/

Todo
- #169 
- Remove uneeded rollup plugins & externals
  - Do we still need terser? 
- Make sure correct types are generated and bundled
- Verify package content with https://publint.dev/

### PR DESCRIPTION
**Pull request by Barsnes**
Resolves #861

### CODE REVIEW

**Comment by Barsnes** [comment]
We are now compiling with `tsc`, and then running rollup to add tree shaking. 
This is the final output when using the library in `create-react-app` and importing **only** `Button` and `Accordion`
image

**Review by mimarz — CHANGES_REQUESTED** [review]
Nice work! 

Made some comments on stuff and suggest trying to also remove the dts plugin as I don't think its needed anymore?

Looking at what rollup does at this point, maybe we won't need it anymore in future as its not doing much now.

I think we can also remove `@rollup/plugin-json` and `@rollup/plugin-image`, but suggest a separate issue on that as to not interfer now.

File: packages/react/rollup.config.js [hunk 1]
```diff
@@ -1,32 +1,36 @@
 import resolve from '@rollup/plugin-node-resolve';
 import commonjs from '@rollup/plugin-commonjs';
-import typescript from '@rollup/plugin-typescript';
 import json from '@rollup/plugin-json';
 import image from '@rollup/plugin-image';
 import dts from 'rollup-plugin-dts';
```
[the remark this answers is not in the document]
Comment (Barsnes) [reply packages/react/rollup.config.js:5]: I think we can do that yes, good catch.
testing..

**Review by mimarz — APPROVED** [review]
Awesome work! 👏 Feels like a big load was shaken off! <3 

I would like to test using something like `bun` or `esbuild` to speed up build time, but we can save to for another time once the rest of our planned changes are done :)

**Comment by mimarz** [comment]
resolved #169

**Pull request merged**